src/hotspot/share/runtime/thread.hpp

```diff
@@ -500,9 +500,9 @@ class Thread: public ThreadShadow {
     return is_in_stack_range_incl(adr, os::current_stack_pointer());
   }
 
-  // Sets this thread as starting thread. Returns failure if thread
+  // Sets the argument thread as starting thread. Returns failure if thread
   // creation fails due to lack of memory, too many threads etc.
-  bool set_as_starting_thread();
+  static bool set_as_starting_thread(JavaThread* jt);
```

Comment:
Why change this? Overall I don't understand the "starting thread" changes.

Reply:
I changed this with the argument that it seemed very unbalanced to have a non-static instance method available on all JavaThreads for calling t->set_as_starting_thread(), for setting a static field, a static field which further is only set once, by the main thread during startup in Threads::create_VM().

The assert method is_starting_thread(Thread*t) is for asserting the special assignment of 1 to the primordial thread.

Reply:
Okay so nothing inherently related to the critical section removal, just an API cleanup.

Reply:
Actually the more I look at this the more I dislike the fact `set_as_starting_thread` also calls `os::create_main_thread`. Seems to me that should be done directly by the caller, and if successful we call `set_as_starting_thread`. But that is a different cleanup.

Reply:
Just FTR `set_as_starting_thread` originally only called `os::create_main_thread` so why it even existed is a bit of a mystery. The actual `_starting_thread` variable was something I added under JDK-8214097.